### ISSUE TITLE
linter: --format=github should include filenames

### ISSUE THREAD

**Issue by nihalgonsalves** [oxc-project/oxc#20492]
The annotations work well, and are fine if it's a small diff with warnings displayed inline.

In our case, we enable new rules as warnings with `--max-warnings` and therefore have a lot of existing warnings in the codebase. It's impossible to see where exactly a new warning is introduced from the GitHub Actions output:

```
Warning: Usage of Array index in keys is not allowed
Warning: Usage of Array index in keys is not allowed
Warning: Forbidden non-null assertion.
Warning: Forbidden non-null assertion.
Warning: Usage of Array index in keys is not allowed
Warning: Usage of Array index in keys is not allowed
Warning: Disabled test
Warning: Disabled test
Warning: Unexpected `any`. Specify a different type.
Warning: Unexpected `any`. Specify a different type.
Warning: Usage of Array index in keys is not allowed
Warning: Error: Calling setState synchronously within an effect can trigger cascading renders
```

(Image description: the above text shown as a screenshot in the GitHub Actions log above, with colouring but no filenames)

<img width="1664" height="550" alt="" src="https://github.com/user-attachments/assets/a3595a21-23e4-439c-84ea-c258fd8cbeba" />

Post-processing fixes the problem for now, and arguably GitHub should also show the filename and line number in the log view, but it would be nice to fix this on oxc's side until then.

---

My workaround:

```sh
oxlint --format=github | node --input-type=module -e '
    import readline from "node:readline";

    const rl = readline.createInterface({
      input: process.stdin,
      crlfDelay: Infinity,
    });

    rl.on("line", (line) => {
      const m = line.match(/^::(warning|error) (.+?)::(.*)$/);
      if (!m) {
        console.log(line);
        return;
      }

      const [, level, meta, msg] = m;

      const file  = meta.match(/(?:^|,)file=([^,]+)/)?.[1];
      const row   = meta.match(/(?:^|,)line=([0-9]+)/)?.[1];
      const col   = meta.match(/(?:^|,)col=([0-9]+)/)?.[1];
      const title = meta.match(/(?:^|,)title=([^,]+)/)?.[1];

      const loc = [file, row, col].filter(Boolean).join(":");

      const prefix = [
        title && `[${title}]`,
        loc && `[${loc}]`
      ].filter(Boolean).join(" ");

      console.log(`::${level} ${meta}::${prefix} ${msg}`.trim());
    });'
```

**Comment by camc314** [comment]
🫠 I wondered why it didn't show up

**Comment by csvn** [comment]
I also wish for the same, if the `--format=github` reporter included filenames and line:col, it would mitigate the need for workarounds like running `oxfmt` twice or piping to JSON (also posted in https://github.com/oxc-project/oxc/issues/21574#issuecomment-4928788761)
